### PR TITLE
fix: local claim validation

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes QCK-29

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## 3. Implementation details

- fixes off by one issue in epoch self consensus state generation that breaks local claim validation
- adds test cases covering local claims with proof validation enabled
- bump cosmos-sdk and tendermint

## 4. How to test/use

- submission of a claim for qassets on the local chain will now succeed.

### CODE REVIEW

File: utils/coins.go [hunk 1]
```diff
@@ -17,14 +17,15 @@ func CoinFromRequestKey(query []byte, accAddr sdk.AccAddress) (sdk.Coin, error)
...
 
 func DenomFromRequestKey(query []byte, accAddr sdk.AccAddress) (string, error) {
-	idx := bytes.Index(query, accAddr)
-	if idx == -1 {
-		return "", errors.New("AccountBalanceCallback: invalid request query")
-	}
-	denom := string(query[idx+len(accAddr):])
-	if err := sdk.ValidateDenom(denom); err != nil {
+	balancesStore := query[1:]
+	accAddr2, denom, err := banktypes.AddressAndDenomFromBalancesStore(balancesStore)
```
Comment (aljo242) [inline utils/coins.go:21]: Let's add a test for this function.  Should cover:
- valid case,
- error in `AddressAndDenomFromBalancesStore`
- not matching account addressses

File: x/participationrewards/keeper/submodule_liquid.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/bech32"
-
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	"github.com/ingenuity-build/quicksilver/utils"
 	"github.com/ingenuity-build/quicksilver/x/participationrewards/types"
```
Comment (aljo242) [inline x/participationrewards/keeper/submodule_liquid.go:11]: format imports

File: cmd/quicksilverd/forceprune.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/syndtr/goleveldb/leveldb/opt"
 	"github.com/syndtr/goleveldb/leveldb/util"
 	tmstore "github.com/tendermint/tendermint/store"
-	tmdb "github.com/tendermint/tm-db"
+	tmdb "github.com/cometbft/cometbft-db"
 
 	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/tendermint/tendermint/config"
```
Comment (aljo242) [inline cmd/quicksilverd/forceprune.go:18]: format imports

**Review by ajansari95 — COMMENTED** [review]
golangci-lint failing otherise looks good

File: cmd/quicksilverd/forceprune.go (hunk 3 above)
Comment (joe-bowman) [reply cmd/quicksilverd/forceprune.go]: done

File: x/participationrewards/keeper/submodule_liquid.go (hunk 2 above)
Comment (joe-bowman) [reply x/participationrewards/keeper/submodule_liquid.go]: done

File: utils/coins.go (hunk 1 above)
Comment (joe-bowman) [reply utils/coins.go]: added

**Review by ajansari95 — APPROVED** [review]
LGTM

**Review by aljo242 — APPROVED** [review]

**Pull request merged**